great_expectations/agent/agent_server_session.py

```diff
@@ -0,0 +1,27 @@
+import pydantic
+from requests import Session
+
+from great_expectations.core.http import create_session
+
+
+class GxAgentServerSessionConfig(pydantic.BaseSettings):
+    gx_cloud_base_url: str = "https://api.greatexpectations.io"
+    gx_cloud_organization_id: str
+    gx_cloud_access_token: str
+
+
+class AgentServerSession:
+    def __init__(self, config: GxAgentServerSessionConfig):
+        self._config = config
+
+    def session(self) -> Session:
+        return create_session(access_token=self._config.gx_cloud_access_token)
+
+    def build_url(self, resource_endpoint: str) -> str:
```

Comment:
I think we have enums for the resource endpoints.

Reply:
https://github.com/great-expectations/great_expectations/blob/70198abacaf7f0713400f64b87c8f51a8f04b137/great_expectations/data_context/cloud_constants.py#L19-L31

```suggestion
    def build_url(self, resource_endpoint: GXCloudRESTResource) -> str:
```

Reply:
The use case for this session wrapper is to allow agent actions to easily query the Cloud backend directly. This would add an unnecessary dependency on OSS code - I would expect that pretty much every time we use this, the endpoint we're hitting is not used by OSS (outside of the agent)

Reply:
I don't understand the point, or maybe I just disagree with it.
The agent already depends on OSS code, and you are doing extra defensive coding to account for the fact that you aren't using something like this enum for `resource_endpoint`.

Reply:
the set of endpoints the Agent uses has little overlap with the set of endpoints that OSS uses (since, if OSS already is using an endpoint, the agent should just rely on OSS functionality). Given that the agent will be refactored to its own repo, this suggestion would mean we need to make a PR into OSS before we could update the agent to use some new mercury endpoint.

Reply:
My understanding is that the only endpoints the agents are currently using are defined in this OSS enum. That's why I suggested using it.
If that's not true, then please create a new enum.